cargo-dist-schema/src/lib.rs

```diff
@@ -159,6 +159,46 @@ pub struct GithubCiInfo {
     pub external_repo_commit: Option<String>,
 }
 
+/// A github action workflow step
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "kebab-case")]
+pub struct GithubJobStep {
```

Comment:
This repesents the full yml object allowed in [the gh docs](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context) though I am not sure if supporting all items is desired. For example having support for `run` may not be something worth supporting in favor of having users develop gh actions (which can be hosted in the repo directly).